### PR TITLE
Composer control feature

### PR DESCRIPTION
**Pull request by jumagura**
This PR is adds two features to improve topic management when a custom wizard replaces the composer:

1) Backend control to block direct topic creation in specified categories.

image

2) A frontend setting to hide categories from the composer dropdown.

image

Awaiting review from @angusmcleod

### CODE REVIEW

File: lib/custom_wizard/extensions/topic_extension.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        :base,
+        I18n.t('wizard.error_messages.wizard_replacing_composer')
+      )
+    end
+  end
+
+  def wizard_replacing_composer?(category_id)
+    return false unless category_id
+
+    category = Category.find(category_id)
```
Comment (angusmcleod) [inline lib/custom_wizard/extensions/topic_extension.rb:20]: The category model instance is already available on both the post and topic models so we don't need to look it up again here.

**Review by angusmcleod — CHANGES_REQUESTED** [review]

File: lib/custom_wizard/extensions/topic_extension.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module CustomWizardTopicExtension
```
[the remark this answers is not in the document]
Comment (jumagura) [reply lib/custom_wizard/extensions/topic_extension.rb]: done :)

File: lib/custom_wizard/extensions/topic_extension.rb (hunk 1 above)
Comment (jumagura) [reply lib/custom_wizard/extensions/topic_extension.rb]: done

**Comment by jumagura** [comment]
frontend setting to hide categories was removed and the category will be hidden from dropdown by default

File: lib/custom_wizard/extensions/topic_extension.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+DiscourseEvent.on(:before_create_topic) do |topic_params, user|
```
Comment (angusmcleod) [inline lib/custom_wizard/extensions/topic_extension.rb:3]: @jumagura It's better to put this directly in the plugin.rb and without the `DiscourseEvent` prefix, i.e. to use the server-side plugin api available in the `plugin.rb`.

**Review by angusmcleod — CHANGES_REQUESTED** [review]
@jumagura We're getting close! Just one small change and this will be ready.

File: lib/custom_wizard/extensions/topic_extension.rb (hunk 3 above)
Comment (jumagura) [reply lib/custom_wizard/extensions/topic_extension.rb]: Done!

**Comment by jumagura** [comment]
Changes requested done 😃

File: test/javascripts/acceptance/category-chooser-initializer-test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+import { test } from "qunit";
+
+acceptance("CategoryChooser", function (needs) {
+  needs.user();
+  needs.settings({
+    allow_uncategorized_topics: false,
+  });
+
+  test("does not display category with custom_wizard_hide_from_composer set to 't'", async function (assert) {
```
Comment (angusmcleod) [inline test/javascripts/acceptance/category-chooser-initializer-test.js:12]: @jumagura Is this testing anything now that the setting is removed?

Comment (jumagura) [reply test/javascripts/acceptance/category-chooser-initializer-test.js]: That test needed refactoring. Now it reviews the composer and checks if filtering is working correctly

**Comment by angusmcleod** [comment]
@jumagura just ce hiking that the PR is not intended to do the second item anymore?

> A frontend setting to hide categories from the composer dropdown.

**Comment by jumagura** [comment]
that is correct. that setting is removed

**Review by angusmcleod — APPROVED** [review]

**Pull request merged**